src/services/Editor/Editor.js

```diff
@@ -747,3 +737,40 @@ export const firstTruthyValue = function (object, acceptableKeys) {
   const matchingKey = _find(acceptableKeys, (key) => object[key]);
   return matchingKey ? object[matchingKey] : undefined;
 };
+
+/**
+ * Constructs the appropriate editor URI based on the editor type
+ */
+export const constructEditorUri = (
+  editor,
+  task,
+  mapBounds,
+  options = {},
+  taskBundle = null,
+  comment = null,
+) => {
+  if (!task) return null;
+
+  switch (editor) {
+    case Editor.id:
+      return constructIdURI(task, mapBounds, options, taskBundle, comment);
+    case Editor.level0:
+      return constructLevel0URI(task, mapBounds, options, taskBundle, comment);
+    case Editor.rapid:
+      return constructRapidURI(task, mapBounds, options, comment);
+    default:
+      return null;
```

Comment:
Nit: I think this should throw; it's a programmer error to call this function with an editor ID other than one of the supported cases.